13.0/s2i/bin/assemble

```diff
@@ -171,7 +172,41 @@ function configure_proxy() {
   xml="$xml\
        </proxy>"
     sed -i "s|<!-- ### configured http proxy ### -->|$xml|" $HOME/.m2/settings.xml
+    return
   fi
+
+  # if http(s)_proxy is set, use that.
+  proxy=${HTTPS_PROXY:-${https_proxy:=${HTTP_PROXY:=$http_proxy}}}
+  noproxy=${NO_PROXY:-$no_proxy}
+  if [ -n $proxy ]; then
+    proxyhost=$(echo $proxy | cut -d : -f 1,2)
+    proxyport=$(echo $proxy | cut -d : -f 3)
+    if [[ `echo $proxy | grep -i https` ]]; then
+      proxyport=${proxyport:-443}
+    else
+      proxyport=${proxyport:-80}
+    fi
+
+    xml="<proxy>\
+         <id>genproxy</id>\
+         <active>true</active>\
+         <protocol>http</protocol>\
+         <host>$proxyhost</host>\
+         <port>$proxyport</port>"
+    if [ -n "$HTTP_PROXY_USERNAME" -a -n "$HTTP_PROXY_PASSWORD" ]; then
```

Comment:
Should this be HTTPS_ ?

Reply:
no, we never had separate http+https variables for username/password.  it really should have been just be "proxy_username" and "proxy_password" probably, but i don't want to force people to change.

Reply:
I had the same gut feeling - rename would require customers to change if they were passing this in as an env var.